scripts/ci/jenkins_diff.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/sh -le
+
+# This is an Jenkins-specific script that runs infracost on the current branch.
+# Usage docs: https://www.infracost.io/docs/integrations/
+
+fix_env_vars () {
+    # Jenkins has problems with envs case sensivity
+    iac_path=${iac_path:-$IAC_PATH}
+    terraform_plan_flags=${terraform_plan_flags:-$TERRAFORM_PLAN_FLAGS}
+    terraform_workspace=${terraform_workspace:-$TERRAFORM_WORKSPACE}
+    usage_file=${usage_file:-$USAGE_FILE}
+    config_file=${config_file:-$CONFIG_FILE}
+    fail_condition=${fail_condition:-$FAIL_CONDITION}
+}
+
+process_args () {
+  # Set variables based on the order for GitHub Actions, or the env value for other CIs
+  iac_path=${1:-$iac_path}
+  terraform_plan_flags=${2:-$terraform_plan_flags}
+  terraform_workspace=${3:-$terraform_workspace}
+  usage_file=${4:-$usage_file}
+  config_file=${5:-$config_file}
+  fail_condition=${7:-$fail_condition}
+
+  # Validate fail_condition
+  if ! echo "$fail_condition" | jq empty; then
+    echo "Error: fail_condition contains invalid JSON"
+  fi
+
+  # Set defaults
+  if [ ! -z "$fail_condition" ] && [ "$(echo "$fail_condition" | jq '.percentage_threshold')" != "null" ]; then
+    fail_percentage_threshold=$(echo "$fail_condition" | jq -r '.percentage_threshold')
+  fi
+  fail_percentage_threshold=${fail_percentage_threshold:--1}
+  INFRACOST_BINARY=${INFRACOST_BINARY:-infracost}
+
+  # Export as it's used by infracost, not this script
+  export INFRACOST_LOG_LEVEL=${INFRACOST_LOG_LEVEL:-info}
+  export INFRACOST_CI_DIFF=true
+
+  if [ ! -z "$GIT_SSH_KEY" ]; then
+    echo "Setting up private Git SSH key so terraform can access your private modules."
+    mkdir -p .ssh
+    echo "${GIT_SSH_KEY}" > .ssh/git_ssh_key
+    chmod 600 .ssh/git_ssh_key
+    export GIT_SSH_COMMAND="ssh -i $(pwd)/.ssh/git_ssh_key -o 'StrictHostKeyChecking=no'"
+  fi
+
+}
+
+build_breakdown_cmd () {
+  breakdown_cmd="${INFRACOST_BINARY} breakdown --no-color --format json"
+
+  if [ ! -z "$iac_path" ]; then
+    breakdown_cmd="$breakdown_cmd --path $iac_path"
+  fi
+  if [ ! -z "$terraform_plan_flags" ]; then
+    breakdown_cmd="$breakdown_cmd --terraform-plan-flags \"$terraform_plan_flags\""
+  fi
+  if [ ! -z "$usage_file" ]; then
+    breakdown_cmd="$breakdown_cmd --usage-file $usage_file"
+  fi
+  if [ ! -z "$config_file" ]; then
+    breakdown_cmd="$breakdown_cmd --config-file $config_file"
+  fi
+  echo "$breakdown_cmd"
+}
+
+build_output_cmd () {
+  breakdown_path=$1
```

Comment:
@aliscott I saw this line in diff.sh and atlantis_diff.sh too, is it redundant since it's not used in this function or am I missing something?

Reply:
Removed it from this script but left the Atlantis for another pr.